### PR TITLE
Fix export import

### PR DESCRIPTION
**Pull request by sphuber**
Opening this pull request to be able to collect information on things that remain to be done before we can merge it into the release branch.

### CODE REVIEW

**Comment by giovannipizzi** [comment]
Do we really want to keep "UNSPECIFIED" as a type of link? I think we should remove it from anyway in the code (see e.g. aiida_core/aiida/common/links.py but then also used as a default parameter in a number of places) an maybe the migration scripts should also check for this type of link beside `null` or `''`?

File: aiida/backends/djsite/db/subtests/nodes.py [hunk 1]
```diff
@@ -96,7 +96,8 @@ def test_links_and_queries(self):
 class TestTransitiveClosureDeletionDjango(AiidaTestCase):
```
Comment (giovannipizzi) [inline aiida/backends/djsite/db/subtests/nodes.py]: These tests should now be backend-independent, right? Can we move them to the set of backend-independent tests?

File: aiida/backends/sqlalchemy/tests/nodes.py [hunk 2]
```diff
@@ -22,8 +22,8 @@ class TestTransitiveClosureDeletionSQLA(AiidaTestCase):
     """
     def test_creation_and_deletion(self):
```
Comment (giovannipizzi) [inline aiida/backends/sqlalchemy/tests/nodes.py]: Same comment here - now these tests and the one of Django should be identical?

File: aiida/backends/djsite/db/subtests/nodes.py (hunk 1 above)
Comment (lekah) [reply aiida/backends/djsite/db/subtests/nodes.py]: They should be, but there's no way to delete links backend independently. This kind of test relies on creating links and deleting them again, and this is done via the backend-specific models.

File: aiida/backends/sqlalchemy/tests/nodes.py (hunk 2 above)
Comment (lekah) [reply aiida/backends/sqlalchemy/tests/nodes.py]: same answer here

**Comment by lekah** [comment]
Regarding @giovannipizzi 's comment above:
This pull-request is about fixing certain very important issues related to import-export breaking with the new link-types (and dbpath-removal). The link-type "UNSPECIFIED" is a valid link-type based on our discussions. If a user (because that link-type is an input given by the user) selects that link-type, we should not overwrite it in a migration, since the migration only migrates an old-database without explicit link-type (null or '') to the new explicit link-types.

**Pull request merged**